### ISSUE TITLE
Using modes in code connect

### ISSUE THREAD

**Issue by MDMann** [figma/code-connect#6]
Hi,
first of all awesome work!
I attended your announcement Zoom meeting and am in awe of the possibility of connecting code snippets to each component and its instances. While watching the demo, I was wondering if it would be possible to also connect modes to code. I asked this question in the meeting but was asked to share more information here.

So basically we set up our components with component-level variables and modes. This means we only have one instance per component and handle the rest with modes.
For example, our "Button" component is just one instance, and the different variants like "Primary", "Secondary" and"Tertiary" are three modes we created in a "Button - Variants" collection. Same for "Sizes" and "States".

In order for us to use code connect it would be great if there was a way to tell the code if for example the mode "Button - Variant: Primary" is selected and map the correct prop to the button in the code.

Let me know if you need more information or if anything is unclear in my explanation.
I would be happy to hear back from you!
Thank you!

**Comment by RoxanneGux** [comment]
Hey MDMANN, I wanted to chime in here, because a few months ago I was going to refactor our Button components to utilize modes instead of variants as you say you are doing. 

I was having some issues and was in conversation with Figma Support. They essentially told me NOT to do this ; to stick with variants for these states of the button, not to use MODES.  They gave me this video to watch. This [link starts at 5:45](https://youtu.be/RXQ8XVirzjs?t=345)

The whole video is worth watching though.

**Comment by RoxanneGux** [comment]
Yes @MDMann  I see. And the "blow up" you describe is what Jake Albaugh showed an example of with the buttons in his live demo- I think it was "Many to one" I laughed because it looks like ours- I attached a screen shot :) 
<img width="985" alt="Screenshot 2024-04-16 at 2 56 59 PM" src="https://github.com/figma/code-connect/assets/104155621/477327f2-96ea-4e49-a859-9c94e19bd4f2">

Good Luck. I hope figma is able to accommodate your request.

**Comment by tomduncalf-figma** [comment]
Hey @MDMann, thanks for the feedback and happy to hear you like Code Connect! This is an interesting idea, and I've raised it for discussion with the team, so we will keep you posted.

Just to verify my understanding, you’d like to be able to do something similar to [Variant Restrictions](https://github.com/figma/code-connect/blob/main/react/README.md#variant-restrictions), but based on the mode of the instance instead?

**Comment by MDMann** [comment]
Hi @tomduncalf-figma, thank you for answering!

I'm not 100% sure if Variant Restrictions are what we need for it.

Let's say a designer is using our button component and configurated it like this:

**Component props:**
- Text: My awesome Button

**Layer modes:**
- Variant: Primary
- Size: M
- Content: Text + Icon

This is the code that should be returned when devs inspect said button:
```
<Button
  iconPosition="right"
  onClick={}
  size="m"
  variant="primary"
>
  My awesome button
</Button>
```

To map everything correctly we would need to set up dynamic code snippets, but instead of referencing component props, we would reference the modes.
```
import figma from '@figma/code-connect'

figma.connect(Button, 'https://...', {
  props: {
    text: figma.string('Text'), <---- This comes from a prop
    variant: figma.enum('Type', { <---- This comes from a collection and mode
      Primary: 'primary',
      Secondary: 'secondary',
      Tertiary: 'tertiary',
    }),
    size: figma.enum('Size', { <---- This comes from a collection and mode
      M: 'm',
      L: 'l',
      S: 's',
    }),
    iconPosition: figma.enum('Content', { <---- This comes from a collection and mode
      Text: 'undefined',
      Icon + Text: 'left',
      Text + Icon: 'right',
    }),
  },
  example: ({ text, iconPosition, size, variant }) => {
    return (
      <Button
        iconPosition={iconPosition}
        onClick={}
        size={size}
        variant={variant}
      >
        {text}
      </Button>
    )
  },
})
```

I hope I used the code snippets correctly and this makes my explanation earlier a bit easier to understand.
Thanks again for answering and keep up the great work!

**Comment by tomduncalf-figma** [comment]
I see, thanks for explaining. We'll discuss this as a team and follow up when we've done so!

**Comment by alisonjoseph** [comment]
@tomduncalf-figma wanted to follow up on this as its something we're running into now that we have AI versions of many of our components. Our designers have set these up as variable modes in Figma. If there was a way to somehow use this as a variant restriction that would be helpful. 🙏
![Screenshot 2024-09-04 at 9 49 36 AM](https://github.com/user-attachments/assets/8cd23c0f-b86d-4a87-804f-36976c0ce61d)
Tile component without/with AI presence.
![Screenshot 2024-09-04 at 9 50 54 AM](https://github.com/user-attachments/assets/4836b7db-7899-456d-8348-d139d9f19c2e)

**Comment by slees-figma** [comment]
Hey @alisonjoseph! Thanks for your interest, this isn't something that we've prioritised yet but definitely still thinking about.

**Comment by mcdonaldp2** [comment]
Hey @slees-figma any sort of update on when this might be prioritized?
